src/Acuminator/Acuminator.Analyzers/Analyzers/DAC/MethodsUsageInDacAnalyzer.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using Acuminator.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+
+namespace Acuminator.Analyzers
+{
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public class MethodsUsageInDacAnalyzer : PXDiagnosticAnalyzer
+	{
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+			ImmutableArray.Create
+			(
+				Descriptors.PX1031_DacCannotContainInstanceMethods,
+				Descriptors.PX1032_DacPropertyCannotContainMethodInvocations
+			);
+
+		internal override void AnalyzeCompilation(CompilationStartAnalysisContext compilationStartContext,
+			PXContext pxContext)
+		{
+			compilationStartContext.RegisterSyntaxNodeAction(syntaxContext => AnalyzeDacDeclaration(syntaxContext, pxContext),
+				SyntaxKind.ClassDeclaration);
+		}
+
+		private void AnalyzeDacDeclaration(SyntaxNodeAnalysisContext syntaxContext, PXContext pxContext)
+		{
+			ClassDeclarationSyntax classDeclaration = (ClassDeclarationSyntax) syntaxContext.Node;
+			INamedTypeSymbol typeSymbol =
+				syntaxContext.SemanticModel.GetDeclaredSymbol(classDeclaration, syntaxContext.CancellationToken);
+
+			if (typeSymbol != null && typeSymbol.IsDacOrExtension(pxContext))
+			{
+				foreach (SyntaxNode node in classDeclaration.DescendantNodes(
+					n => !(n is ClassDeclarationSyntax) || IsDacOrExtension(n, pxContext, syntaxContext.SemanticModel, syntaxContext.CancellationToken)))
+				{
+					syntaxContext.CancellationToken.ThrowIfCancellationRequested();
+
+					if (node is MethodDeclarationSyntax method)
+					{
+						AnalyzeMethodDeclarationInDac(method, syntaxContext, pxContext);
+					}
+					else if (node is PropertyDeclarationSyntax property)
+					{
+						AnalyzeMethodInvocationInDacProperty(property, syntaxContext, pxContext);
+					}
+				}
+			}
+		}
+
+		private void AnalyzeMethodInvocationInDacProperty(PropertyDeclarationSyntax property,
+			SyntaxNodeAnalysisContext syntaxContext, PXContext pxContext)
+		{
+			foreach (SyntaxNode node in property.DescendantNodes())
+			{
+				syntaxContext.CancellationToken.ThrowIfCancellationRequested();
+
+				if (node is InvocationExpressionSyntax invocation)
```

Comment:
@dropsonic, imho there is a little inconsistency here - we search for method invocations, constructor invocations, but skip property invocations. What about the property in some helper class?